site/events.njk

```diff
@@ -2,4 +2,16 @@
 disable_algolia: true
 permalink: "/events.json"
 ---
-{{ collections.currentEvents.concat(collections.pastEvents) | dump(2) | safe }}
+[
+  {% for event in collections.currentEvents.concat(collections.pastEvents)  %}
+    {% set html %}
+      {% include 'partials/event-card.njk' %}
+    {% endset %}
+    {
+      "id": {{event.id|dump|safe}},
+      "html": {{ html|minifyHtml|dump|safe}},
+      "date": {{ event.date|dump|safe }},
+      "isPastEvent": {{ event.isPastEvent|dump|safe }}
+    } {% if loop.last === false %},{% endif %}
+  {% endfor %}
+]
```

Comment:
This all looks sus to me 😅  I know one-off filters are also not ideal, but putting this logic into a `prerenderEvents` filter or something like this appears cleaner to me. @devnook, WDYT?

Reply:
Is it possible to render templates/partials in filters? I wasn't able to find an API allowing that, or were you thinking something else?

Reply:
Something like for the partial shortcode might work. But I think one might also be able to extract the nunjucks engine from a shortcode context to doe stuff with it, but need to dig around. Let's wait for Ewa's opinion on this 🙂

Reply:
Best would be to prepare html in the events.js file, where the rest of json is generated. BUT njk engine is not available there yet. So we have to render it in the events.njk template. Wheter we do it with a filter, shortode or include, is't all the same I think. Storing it in variable is as good as any other way, just please double check if it's not escaping things twice.